lib/standalone-benches/benches/channels.rs

```diff
@@ -0,0 +1,145 @@
+use std::{time::Duration, usize};
+
+use criterion::{
+    async_executor, black_box, criterion_group, criterion_main, BenchmarkId, Criterion, Throughput,
+};
+use futures03::{SinkExt, StreamExt};
+
+criterion_group!(benches, benchmark);
+criterion_main!(benches);
+
+fn benchmark(c: &mut Criterion) {
+    let mut group = c.benchmark_group("channels/data_size");
+    for &(batches, batch_size, time_mul) in &[
+        (1_000, 1, 1),
+        (1_000, 10, 1),
+        (1_000, 100, 1),
+        (100_000, 1, 1),
+        (100_000, 10, 1),
+        (100_000, 100, 1),
+    ] {
+        group.noise_threshold(0.01);
+        group.measurement_time(Duration::from_secs(10 * time_mul));
+        group.warm_up_time(Duration::from_secs(3 * time_mul));
+        group.throughput(Throughput::Elements(batches as u64 * batch_size as u64));
+        group.bench_with_input(
+            BenchmarkId::new("tokio 0.2", format!("{}x{}", batches, batch_size)),
+            &(batches, batch_size),
+            |b, &(batches, batch_size)| {
+                b.to_async(async_executor::FuturesExecutor).iter_with_setup(
+                    || {
+                        (
+                            tokio02::sync::mpsc::channel(1),
+                            make_data(batches, batch_size),
+                        )
+                    },
+                    |((tx, rx), input)| run_tokio02(tx, rx, input),
```

Comment:
Small nit, I might have just embedded the `run_x` functions in the benchmark. I'm not sure much is gained by splitting out those functions given they are only used in one place.

Reply:
Ah, right. I wanted to see if we macro them out or generalize, but apparently, they are very specific.
I'll do another pass on making things more compact.

Reply:
Just remembered why I used those - so, the functions are `async`, so each returns a future, and it works with stable rust. The async closures aren't stable yet though, and we'd need those if we want to inline the async calls. Unless I'm not mistaken.

Reply:
I could certainly be missing something, but I think async closures are stable? Using `async move { }` here seems to compile and run fine:

```diff
diff --git a/lib/standalone-benches/benches/channels.rs b/lib/standalone-benches/benches/channels.rs
index 730b0c500..d56b82f67 100644
--- a/lib/standalone-benches/benches/channels.rs
+++ b/lib/standalone-benches/benches/channels.rs
@@ -33,7 +33,13 @@ fn benchmark(c: &mut Criterion) {
                             make_data(batches, batch_size),
                         )
                     },
-                    |((tx, rx), input)| run_tokio02(tx, rx, input),
+                    |((mut tx, mut rx), input)| async move {
+                        for item in input {
+                            tx.send(item).await.unwrap();
+                            let data = rx.recv().await.unwrap();
+                            process_data(data.into_iter());
+                        }
+                    },
                 );
             },
         );
```